core/insertion_marker_manager.ts

```diff
@@ -333,51 +330,39 @@ export class InsertionMarkerManager {
    * Whether the previews (insertion marker and replacement marker) should be
    * updated based on the closest candidate and the current drag distance.
    *
-   * @param candidate An object containing a local connection, a closest
-   *     connection, and a radius.  Returned by getCandidate_.
+   * @param newCandidate A new candidate connection that may replace the current
+   *     best candidate.
    * @param dxy Position relative to drag start, in workspace units.
    * @returns Whether the preview should be updated.
    */
-  private shouldUpdatePreviews_(
-      candidate: CandidateConnection, dxy: Coordinate): boolean {
-    const candidateLocal = candidate.local;
-    const candidateClosest = candidate.closest;
-    const radius = candidate.radius;
-
-    // Found a connection!
-    if (candidateLocal && candidateClosest) {
-      // We're already showing an insertion marker.
-      // Decide whether the new connection has higher priority.
-      if (this.localConnection_ && this.closestConnection_) {
-        // The connection was the same as the current connection.
-        if (this.closestConnection_ === candidateClosest &&
-            this.localConnection_ === candidateLocal) {
-          return false;
-        }
-        const xDiff =
-            this.localConnection_.x + dxy.x - this.closestConnection_.x;
-        const yDiff =
-            this.localConnection_.y + dxy.y - this.closestConnection_.y;
-        const curDistance = Math.sqrt(xDiff * xDiff + yDiff * yDiff);
-        // Slightly prefer the existing preview over a new preview.
-        return !(
-            candidateClosest &&
-            radius > curDistance - config.currentConnectionPreference);
-      } else if (!this.localConnection_ && !this.closestConnection_) {
-        // We weren't showing a preview before, but we should now.
-        return true;
-      } else {
-        console.error(
-            'Only one of localConnection_ and closestConnection_ was set.');
-      }
-    } else {  // No connection found.
+  private shouldUpdatePreviews(
+      newCandidate: CandidateConnection|null, dxy: Coordinate): boolean {
+    if (!newCandidate) {
       // Only need to update if we were showing a preview before.
-      return !!(this.localConnection_ && this.closestConnection_);
+      return !!this.activeCandidate;
```

Comment:
Hooray for guard clauses.  But this and the following could have been one-liners.